### PR TITLE
Upgrade site to Bootstrap 5

### PR DESCRIPTION
**Pull request by cthoyt**
This PR does the following:

1. Massive deletion of unused, vendored Bootstrap 3 and jQuery assets from 2015/2016
2. Massive deletion of unused custom CSS
3. Reorganization of Javascript assets (there was no need to use a maze of includes/imports in Liquid)
4. Upgrade template to use Bootstrap 5

Before merge, I think this warrants some external testing.

## HOW TO TEST

After getting docker running, do the following in your shell

```shell
git clone https://github.com/cthoyt/OBOFoundry.github.io.git@upgrade-bootstrap obofoundry
cd obofoundry
docker run --rm --volume="$PWD:/srv/jekyll" -p 4000:4000 -it jekyll/jekyll:3.5 jekyll serve
```

Finally, navigate to http://localhost:4000 and see how everything works.

## Screenshots

<details>
<summary>Click to see screenshots</summary>

Screen Shot 2022-10-10 at 01 58 09
Screen Shot 2022-10-10 at 01 58 13
Screen Shot 2022-10-10 at 01 58 23
Screen Shot 2022-10-10 at 01 58 55

</details>

### CODE REVIEW

**Comment by cthoyt** [comment]
@erik-whiting any chance you would be able to review this PR by checking out my branch, running Jekyll (instructions above), and checking out the site?

**Comment by erik-whiting** [comment]
@cthoyt will do, give me a couple of hours

**Comment by cthoyt** [comment]
@matentzn it's ready, I know you're busy this week so we'll see if anyone else wants to take a stab

@egonw cc

**Comment by erik-whiting** [comment]
Is it just me it looks like this for?:
image
On the left is this branch running locally, the right is the live website

I'm not using Docker if that matters, though I can't imagine why it would.

I'm also getting this JavaScript error any time I click a dropdown menu in the top:
```
Bootstrap doesn't allow more than one instance per element. Bound instance: bs.tooltip.
```

Speaking of dropdowns, mine are stacking:
image

These seem like "bootstrap is gone now" errors but again, I'm not running this in Docker at the moment (I have never gotten this running in Docker, see my discussion post about it: https://github.com/OBOFoundry/OBOFoundry.github.io/discussions/2154 ) so maybe it's some local issue with my computer not having bootstrap cached or something.

Functionally, everything works the way it should (search, the grouping buttons, etc) but the front-facing stuff is broken, for me at least.

**Comment by cthoyt** [comment]
@erik-whiting that's so weird, I am also now getting the same error... I wonder if this is because of an issue with the CDN? before, these were vendored, but that is a pain as well and I would rather not go that route

**Comment by cthoyt** [comment]
@erik-whiting I figured out the issue in 92da1f6. There was outdated (maybe initially incorrect) usage of `tooltip` since it wasn't bound to any package. The dropdowns now go away properly on the home page.

This again is a reminder that 1) javascript is the worst and 2) hacky solutions for problems in javascript just burns more developer hours in the long run. The OBO Foundry will have to re-write the home page yet again, I'm sure, sooner than hoped

**Comment by erik-whiting** [comment]
> @erik-whiting I figured out the issue in [92da1f6](https://github.com/OBOFoundry/OBOFoundry.github.io/commit/92da1f67b3b05094b69b08007e1bebbf893cc59a). There was outdated (maybe initially incorrect) usage of `tooltip` since it wasn't bound to any package. The dropdowns now go away properly on the home page.
> 
> This again is a reminder that 1) javascript is the worst and 2) hacky solutions for problems in javascript just burns more developer hours in the long run. The OBO Foundry will have to re-write the home page yet again, I'm sure, sooner than hoped

I've been debating whether to open an issue for creating automated tests that test the website itself. There are good data and request tests in the repo but nothing that scrapes the webpages and checks that no functionality has been broken (something that could've caught the [error I introduced](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/2130) in this [PR changing search functionality](https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2114)). I'm not sure such a collection of tests would be helpful for this issue but it might provide more of a safety net for major upgrades and rewrites in the future.

Anyway, I'm still waiting on this Docker image to build and I will poke at your changes when it's done. Will update then.

**Comment by cthoyt** [comment]
I might want to try and make the subheader text smaller but besides that I think this is looking fine. WRT testing, that might be interesting but I'm wondering if the development time/complexity is a good trade-off for just testing stuff ourselves on any other big changes (or specifically ones that involve changing javascript - overall Jekyll and Bootstrap should be considered stable)

@jamesaoverton can you help us push this through while Nico is out of office?

File: allnews.html
Comment (cthoyt) [reply allnews.html:5]: two things happened here:
1. There was unnecessary nesting of layouts/templates, so I flattened them
2. This isn't linked anywhere, but it is still accessible at https://obofoundry.org/allnews.html. If you think this isn't worth keeping, we can do a follow-up PR for that.

File: assets/js/custom.js
Comment (cthoyt) [reply assets/js/custom.js:90]: This is just how the icon libraries work, the class has CSS associated with it that fills up the `<i>` element with some SVG in a magical way I don't really understand. Why it's specifically `<i>` and not `<span>` I'm not sure about, maybe because it's short?

File: index.html
Comment (cthoyt) [reply index.html:20]: I copied it from the about page. I can delete it if desired. There are for sure a lot instances through the OBO site where there's a shallow description of something on one page that links to a more specific page, that contains only a similar shallow description :/

File: assets/js/custom.js
Comment (cthoyt) [reply assets/js/custom.js:469]: This was the problematic tooltip usage that was causing the navbar tabs not to open and close properly. 

I think what happened here was this was a non-standard way of activating the pretty bootstrap tooltips, so for now, it will use browser-based tool tips. Later we can try and get the bootstrap ones working again, but I found the documentation (https://getbootstrap.com/docs/5.2/components/tooltips/) a bit confusing

Comment (matentzn) [reply assets/js/custom.js:469]: Ok!

File: allnews.html
Comment (matentzn) [reply allnews.html:5]: ohhh ok, i think this should be deleted asap! How did you find this!!

File: assets/js/custom.js
Comment (matentzn) [reply assets/js/custom.js:90]: ok!

File: index.html
Comment (matentzn) [reply index.html:20]: Can we move this to separate PR?

**Comment by matentzn** [comment]
Can we get an updated screenshot of the front page?

File: allnews.html
Comment (cthoyt) [reply allnews.html:5]: both the news page and also these outdated blog posts? Or just the page that displays them?

Comment (cthoyt) [reply allnews.html:5]: so if I delete all of these "news" (which I called blog posts since it's using jekyll's built in post functions) then that won't hold this PR up?

File: index.html
Comment (cthoyt) [reply index.html]: Removed in 41520ac

**Comment by cthoyt** [comment]
@matentzn updated screen shot of the homepage

Screen Shot 2022-10-25 at 19 09 46

File: allnews.html
Comment (cthoyt) [reply allnews.html:5]: would rather do this in a follow-up PR to make merging this one faster

**Comment by nlharris** [comment]
Looks fine, though it's slightly non-optimal that "ID ^" and "Re-Use ^" now wrap.
(Hmm, maybe ID already wrapped? When I look at the current site, ID wraps but Re-Use doesn't.)

**Comment by matentzn** [comment]
There are a number of CSS issues on this screenshot:

- [ ] unordered lists under first level subheadings should be deeper indented under the heading (e.g. bullets under "Participate")
- [ ] The down-arrow for sorting under Re-use and ID should be next to the column header, like in the Title case

Before merging this I would like to know one as well: Do you (@erik-whiting and @cthoyt) commit to fixing (medium to major) CSS issues that happen as a result of the migration (on other pages, etc)?

File: allnews.html
Comment (matentzn) [reply allnews.html:5]: sounds good

**Comment by erik-whiting** [comment]
> Before merging this I would like to know one as well: Do you (@erik-whiting and @cthoyt) commit to fixing (medium to major) CSS issues that happen as a result of the migration (on other pages, etc)?

Yeah sure, just assign me anything that comes up

**Comment by matentzn** [comment]
Great @erik-whiting would you mind dealing with the two items above then we merge this here and see what happens?

**Comment by erik-whiting** [comment]
> Great @erik-whiting would you mind dealing with the two items above then we merge this here and see what happens?

yes but I can't promise a quick resolution, the semester is getting pretty crazy

**Comment by matentzn** [comment]
No, @cthoyt i asked you this here to judge how much testing I should do on this PR. I had two options: 1) elicit commitment from you to fix issues Introduced by this PR and merge fast or 2) not elicit commitments and test, list all newly emerged css issues and have them all be fixed before merging. Both are fine, I just wanted to know which of the two you prefer!

**Comment by cthoyt** [comment]
> No, @cthoyt i asked you this here to judge how much testing I should do on this PR. I had two options: 1) elicit commitment from you to fix issues Introduced by this PR and merge fast or 2) not elicit commitments and test, list all newly emerged css issues and have them all be fixed before merging. Both are fine, I just wanted to know which of the two you prefer!

Sorry I'm being nippy. I'm just unhappy about javascript. I fixed the down arrows in 96a38ed (screenshot below). With respect to the indenting of the lists, I'm pretty sure this is a stylistic choice of Bootstrap 5. Note that there's no additional styling information in the HTML around it. I actually like this change, since it means that the text is all vertically aligned together.

Either way, I still am committed to supporting HTML (and if need be, javascript) maintenance and improvement to the site. Maybe another thing for `roles.yml` to make it more clear that this is a thing that somebody actually has to do?

Screen Shot 2022-10-25 at 23 41 47

**Review by matentzn — APPROVED** [review]

**Review by cmungall — APPROVED** [review]

**Comment by matentzn** [comment]
@cthoyt @erik-whiting Amazing job you two! Aside from a few CSS issues, I don't spot any other problems.. Lets use #2166 to collect any related issues.

**Comment by erik-whiting** [comment]
thanks but I actually didn't do anything more than poke at the updates, this was all @cthoyt's effort

**Pull request merged**
